src/Workspaces/Core/Portable/FindSymbols/SymbolTree/SymbolTreeInfo_Serialization.cs

```diff
@@ -138,14 +137,15 @@ async Task<T> CreateWithLoggingAsync()
 
         public void WriteTo(ObjectWriter writer)
         {
-            writer.WriteString(_concatenatedNames);
-
             writer.WriteInt32(_nodes.Length);
-            foreach (var node in _nodes)
+            foreach (var group in GroupByName(_nodes))
             {
-                writer.WriteInt32(node.NameSpan.Start);
-                writer.WriteInt32(node.NameSpan.Length);
-                writer.WriteInt32(node.ParentIndex);
+                writer.WriteInt32(group.Length);
+                writer.WriteString(group.Span[0].Name);
```

Comment:
nit, consider swapping these.  in practically all other places we write "length" then sequence of items.  so here we would write: name, length, items.   That feels morally better to me than: length, name, items.

Reply:
➡️ Reordered

Reply:
thanks.